[quant][ao_migration] Base package in tests

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #77638
* #77630
* #77627
* #77586
* #77582
* #77581
* __->__ #77064
* #77344
* #77065

Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

Differential Revision: [D36459452](https://our.internmc.facebook.com/intern/diff/D36459452)

Commit:
[quant][ao_migration] Base package in tests

Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

[ghstack-poisoned]

jerryzh168 commented on test/quantization/ao_migration/common.py:
```diff
@@ -1,14 +1,18 @@
-        old_module = importlib.import_module(f'torch.quantization.{package_name}')
-        new_module = importlib.import_module(f'torch.ao.quantization.{package_name}')
+        if base is None:
+            base = 'quantization'
```
str is not mutable in python so it's safe to use as a default arg?

jerryzh168 replied:
same for other two functions

jerryzh168 reviewed (approved):
had an inline nit comment

z-a-f commented:
@z-a-f has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36459452).

Commit:
Update on "[quant][ao_migration] Base package in tests"


Adding a base package as an argument to the testing routines.
That will allow us to test other locations that are being migrated.
For example

```
AOMigrationTestCase._test_package_import('my_mackage', base='quantization')
```

would check if `torch.quantization.my_package` and `torch.ao.quantization.my_package` are the same.

Differential Revision: [D36459452](https://our.internmc.facebook.com/intern/diff/D36459452)

[ghstack-poisoned]
